File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        int n = nums.length;
        PriorityQueue<Integer> pq = new PriorityQueue<> (Collections.reverseOrder());
        int tar = n - k ;
        pq.offer(nums[0]);
        for(int i = 1 ; i < n ; i++){
            if(pq.size() <= tar) pq.offer(nums[i]);
            else{
                if(nums[i] < pq.peek()){
                    pq.poll();
                    pq.offer(nums[i]);
                }
            }
        }
        return pq.peek();
    }
}

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        int n = nums.size();
        priority_queue<int> pq;
        int tar = n - k;
        pq.push(nums[0]);
        for(int i = 1 ; i < n ; i++){
            if(pq.size() <= tar) pq.push(nums[i]);
            else{
                if(pq.top() > nums[i]){
                    pq.pop();
                    pq.push(nums[i]);
                }
            }
        }
        return pq.top();
    }
};